### ISSUE TITLE
why the mAP I validate on coco val data is different from the official reported mAP?

### ISSUE THREAD

**Issue by Scorbinwen** [ultralytics/ultralytics#21598]
I I validate on coco val data, and get mAP as follow:
<img width="1195" height="504" alt="Image" src="https://github.com/user-attachments/assets/f8e7aa2a-e8bc-42d5-b8ea-6057ea115d69" />
which is different from the official reported mAP:
<img width="757" height="514" alt="Image" src="https://github.com/user-attachments/assets/96b1d232-7130-4144-8496-f7bfbb845670" />
1. model: yolo11l.pt download from: yolo11l.pt [huggingface](https://huggingface.co/Ultralytics/YOLO11/blob/main/yolo11l.pt)
2. val hyp parameters:
```yaml
val: True # (bool) validate/test during training
split: val # (str) dataset split to use for validation, i.e. 'val', 'test' or 'train'
save_json: False # (bool) save results to JSON file
save_hybrid: False # (bool) save hybrid version of labels (labels + additional predictions)
conf: # (float, optional) object confidence threshold for detection (default 0.25 predict, 0.001 val)
iou: 0.7 # (float) intersection over union (IoU) threshold for NMS
max_det: 300 # (int) maximum number of detections per image
half: True # (bool) use half precision (FP16)
dnn: False # (bool) use OpenCV DNN for ONNX inference
plots: True # (bool) save plots and images during train/val
```
3. coco dataset: download annoations from http://images.cocodataset.org/annotations/annotations_trainval2017.zip 
and use convert_coco("/path/to/annotations", use_segments=True, use_keypoints=False, cls91to80=True) to yolo format
I will be gratefull if anyone can reply.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Scorbinwen, thank you for reaching out and sharing your results with Ultralytics 🚀! This is an automated response to help get you started—an Ultralytics engineer will follow up soon to assist further.

For differences between your mAP and official reports, there may be various causes such as evaluation settings, dataset versions, or inference parameters. To help us investigate, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) showing your complete validation command or script, environment details, and any custom changes you made. This will help us reproduce and troubleshoot the issue efficiently 🕵️‍♂️.

In the meantime, we recommend reviewing the [Docs](https://docs.ultralytics.com/), especially the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and verifying you are using the latest release.

Join the Ultralytics community for more support! For real-time chat, connect on [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, check out [Discourse](https://community.ultralytics.com/), or join our [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Upgrade

Please make sure you are running the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Ultralytics reports mAP obtained by running COCOEval. You can get that by using `data=coco.yaml`

**Comment by Scorbinwen** [comment]
> Ultralytics reports mAP obtained by running COCOEval. You can get that by using `data=coco.yaml`

Thank you very much，I found it's because I use the wrong imgsz for validation
I write my own val.py as follow:
```python
from pathlib import Path
import sys
import os
import yaml
import argparse
import tempfile

# 设置环境变量和路径
os.environ['MKL_SERVICE_FORCE_INTEL'] = 'True'
FILE = Path(__file__).resolve()
ROOT = FILE.parents[1]
if str(ROOT) not in sys.path:
    sys.path.append(str(ROOT))
ROOT = Path(os.path.relpath(ROOT, Path.cwd()))

from ultralytics import YOLO
from ultralytics.utils.ops import parse_opt

if __name__ == "__main__":
    args = parse_opt()

    with open(args.cfg) as f:
        cfg_params = yaml.safe_load(f)

    command_overrides = {k: v for k, v in vars(args).items()
                         if k in cfg_params.keys() and v != cfg_params.get(k)}
    cfg_params.update(command_overrides)

    model = YOLO(args.model)
    model.val(**cfg_params)
```
**and the default imgsz defined in argparse is [320, 320]:**
```python
def parse_opt():
    base_parser = argparse.ArgumentParser()
    base_parser.add_argument('--model', default="../ultralytics/cfg/models/11/yolo11l.yaml", help='/path/to/yolo.yaml)
    base_parser.add_argument('--cfg', default="../ultralytics/cfg/hyps/hyps.scratch-low.yaml", help='/path/to/hyps.yaml')
    base_parser.add_argument('--imgsz', nargs='+', type=int, default=[320, 320],
                        help='imgsz eg. --imgsz 320 320')
    base_args, unknown = base_parser.parse_known_args()
```

**Comment by Scorbinwen** [comment]
> > Ultralytics reports mAP obtained by running COCOEval. You can get that by using `data=coco.yaml`
> 
> Thank you very much，I found it's because I use the wrong imgsz for validation I write my own val.py as follow:
> 
> from pathlib import Path
> import sys
> import os
> import yaml
> import argparse
> import tempfile
> 
> # 设置环境变量和路径
> os.environ['MKL_SERVICE_FORCE_INTEL'] = 'True'
> FILE = Path(__file__).resolve()
> ROOT = FILE.parents[1]
> if str(ROOT) not in sys.path:
>     sys.path.append(str(ROOT))
> ROOT = Path(os.path.relpath(ROOT, Path.cwd()))
> 
> from ultralytics import YOLO
> from ultralytics.utils.ops import parse_opt
> 
> if __name__ == "__main__":
>     args = parse_opt()
> 
>     with open(args.cfg) as f:
>         cfg_params = yaml.safe_load(f)
> 
>     command_overrides = {k: v for k, v in vars(args).items()
>                          if k in cfg_params.keys() and v != cfg_params.get(k)}
>     cfg_params.update(command_overrides)
> 
>     model = YOLO(args.model)
>     model.val(**cfg_params)
> **and the default imgsz defined in argparse is [320, 320]:**
> 
> def parse_opt():
>     base_parser = argparse.ArgumentParser()
>     base_parser.add_argument('--model', default="../ultralytics/cfg/models/11/yolo11l.yaml", help='/path/to/yolo.yaml)
>     base_parser.add_argument('--cfg', default="../ultralytics/cfg/hyps/hyps.scratch-low.yaml", help='/path/to/hyps.yaml')
>     base_parser.add_argument('--imgsz', nargs='+', type=int, default=[320, 320],
>                         help='imgsz eg. --imgsz 320 320')
>     base_args, unknown = base_parser.parse_known_args()

**Comment by glenn-jocher** [comment]
Great catch! You're absolutely right - image size is critical for validation accuracy. The [official YOLO11 models](https://docs.ultralytics.com/datasets/detect/coco/#coco-pretrained-models) are validated at `imgsz=640`, so using `[320, 320]` would definitely give different mAP results. For accurate comparisons with official benchmarks, always use `imgsz=640` during validation.
